src/daemon/rpmostreed-deployment-utils.cxx

```diff
@@ -767,8 +928,15 @@ rpmostreed_update_generate_variant (OstreeDeployment *booted_deployment,
   if (!rpm_diff_is_empty (&rpm_diff))
     g_variant_dict_insert (dict, "rpm-diff", "@a{sv}", rpm_diff_variant_new (&rpm_diff));
 
-  /* now we look for advisories */
+  gboolean check_container_manifest_diff = FALSE;
+  if (!manifest_diff_is_empty (&manifest_diff))
```

Comment:
I'd use `if (refspectype == rpmostreecxx::RefspecType::Container)` here because if we're in a container flow, we actually *do* want to emit the manifest-diff but as an empty state, right?  That would signal to clients "there is no available update", instead of having the property not be present.

Reply:
I agree that having an the `manifest-diff` key in the dictionary still exist, but be empty if there is no update makes sense. However, I modeled this function after:
```
if (!rpm_diff_is_empty (&rpm_diff))
    g_variant_dict_insert (dict, "rpm-diff", "@a{sv}", rpm_diff_variant_new (&rpm_diff));
```
Which does not create the `rpm-diff` key and is handled on the client side to check if it does or does not exist. Should I aim to follow the same method for the existing code, or just make the changes that you have suggested.

Reply:
Right, following existing code usually makes sense.  I don't honestly know the rationale why the person who wrote the previous code chose to represent "no changes" as "absent property" instead of "zero packages changed diff".

I'm fine if you want to omit the property in the unchanged case too.

But one minor detail here: Having the property always be present allows an external client to reliably know there isn't an update.  Whereas the property also might just be absent for older versions of rpm-ostreed.